### DISCUSSION TITLE
[Client OPC-UA] - Simultaneous management of reading and writing nodes

### DISCUSSION THREAD

**Discussion by Nicrus86** [FreeOpcUa/opcua-asyncio#1613] [Q&A]
Hello to everyone!
On the opcua client side, it is possible to read the changes of a node by subscribing, and at the same time perform write operations on some node ?
My code has the same kind of structure as the --> [client-subscription.py](https://github.com/FreeOpcUa/opcua-asyncio/blob/master/examples/client-subscription.py). Can you suggest a way to manage nodes to write (which do not always coincide with those to read) ?
Thanks in advance *_*

**Comment by AndreasHeine** [comment]
if you want to write in the datachange callback you need to make him async -> the datachange callback can be a coro too 😉
